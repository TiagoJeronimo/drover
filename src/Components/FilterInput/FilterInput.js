import React from 'react';
import { Input } from 'semantic-ui-react';
import './FilterInput.style.scss';

const FilterInput = ({ labelText, inputFieldName, onClickInputField, handleInputChange }) => (
  <div className="filterInput-container">
    <div className="filterInput-label">{labelText}</div>
    <Input
      action={{
        icon: 'search',
        onClick: onClickInputField
      }}
      placeholder="London, Uk"
      name={inputFieldName}
      onChange={(e, { name, value }) => handleInputChange(name, value)}
    />
  </div>
);

export default FilterInput;
