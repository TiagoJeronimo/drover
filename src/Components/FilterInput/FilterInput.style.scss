.filterInput-container {
  width: 100%;
  min-height: 25px;
  margin-bottom: 16px;
}

.filterInput-label {
  margin: 10px 0;
  font-size: 16px;
}
