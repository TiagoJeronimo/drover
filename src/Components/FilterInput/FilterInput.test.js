import React from 'react';
import { mount } from 'enzyme';

import FilterInput from './FilterInput';

it('FilterInput with normal props', () => {
  const props = {
    labelText: 'Location',
    inputFieldName: 'location',
    onClickInputField: jest.fn(),
    handleInputChange: jest.fn()
  };
  const component = <FilterInput {...props} />;
  const html = mount(component).html();
  expect(html).toMatchSnapshot();
});
