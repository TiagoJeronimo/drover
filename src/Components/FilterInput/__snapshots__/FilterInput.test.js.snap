// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterInput with normal props 1`] = `"<div class=\\"filterInput-container\\"><div class=\\"filterInput-label\\">Location</div><div class=\\"ui action input\\"><input placeholder=\\"London, Uk\\" name=\\"location\\" type=\\"text\\"><button class=\\"ui icon button\\"><i aria-hidden=\\"true\\" class=\\"search icon\\"></i></button></div></div>"`;
